### ISSUE TITLE
Hit recursion limit in Firefox

### ISSUE THREAD

**Issue by Robinfr** [Reactive-Extensions/RxJS#1524]
I have a system that has to merge a lot of observables together (1,500 of them) using Rx.Observable.merge. However, doing this, I seem to hit the recursion limit in Firefox. Are the observables being merged recursively?

A fix seems to be to split them up into chunks of 500 and merge those, and the merge the results once more.

What is the correct solution here? And am I right in understanding that the observables get merged recursively?

**Comment by paulpdaniels** [comment]
Stackoverflow is going to get you more responses as its format is more for Q&A. Also unless you are stuck with RxJS < 5 you should post bugs at ReactiveX/rxjs instead. This repo is in maintenance mode and you won't get timely responses.

**Comment by Robinfr** [comment]
Ah unfortunately I am stuck with version 4. Also, I posted it because I feel like this might have to with an implementation detail the developers are more aware of. Are you aware of any changes they've made to the observable merging going from 4 to 5?

**Comment by paulpdaniels** [comment]
I'm not sure I haven't followed that operators development for a while. I am not entirely sure what you are doing in your use case, could you post some code? Is it something like `Observable.merge([...arrayOf1500])`?

**Comment by Robinfr** [comment]
Yes exactly what you mentioned. It's an array of 1500 Rxjs Subject objects.

Op 7 nov. 2017 5:21 p.m. schreef Paul Daniels <notifications@github.com>:

I'm not sure I haven't followed that operators development for a while. I am not entirely sure what you are doing in your use case, could you post some code? Is it something like Observable.merge([...arrayOf1500])?
